### DISCUSSION TITLE
Fine-tuning AraGPT2

### DISCUSSION THREAD

**Discussion by AnwarQarout** [aub-mind/arabert#166] [Q&A]
Hello

For newbies, is there any guide on how to fine-tune AraGPT2 with a dataset that I have?

**Comment by WissamAntoun** [comment]
WHile I didn't write one for AraGPT2, you can use any regular gpt2 tutorial with aragpt2.

here is one https://colab.research.google.com/github/philschmid/fine-tune-GPT-2/blob/master/Fine_tune_a_non_English_GPT_2_Model_with_Huggingface.ipynb 

for best results make sure to preprocess your text dataset first with the preprocessor included in our repo.
